app/assets/javascripts/discourse/app/lib/range-ref.js

```diff
@@ -0,0 +1,42 @@
+export default class RangeRef {
```

Comment:
I think a more explicit name here would be something like: 

virtualElementFromTextRange

Reply:
Also might be better as an exported function, which would hide the class implementation